WalletWasabi/Models/UpdateStatus.cs

```diff
@@ -21,9 +23,9 @@ public UpdateStatus(bool backendCompatible, bool clientUpToDate)
 
 		public bool Equals(UpdateStatus other) => this == other;
 
-		public override int GetHashCode() => ClientUpToDate.GetHashCode() ^ BackendCompatible.GetHashCode();
+		public override int GetHashCode() => (13 * ClientUpToDate.GetHashCode()) ^ (47 * BackendCompatible.GetHashCode()) ^ LegalDocumentsVersion.GetHashCode();
 
-		public static bool operator ==(UpdateStatus x, UpdateStatus y) => y?.ClientUpToDate == x?.ClientUpToDate && y?.BackendCompatible == x?.BackendCompatible;
+		public static bool operator ==(UpdateStatus x, UpdateStatus y) => y?.ClientUpToDate == x?.ClientUpToDate && y?.BackendCompatible == x?.BackendCompatible && y?.LegalDocumentsVersion == x?.LegalDocumentsVersion;
```

Comment:
Use our nice` GetHashCode()` function. 

```suggestion
		public static bool operator ==(UpdateStatus x, UpdateStatus y) => y?.GetHashCode() == x?.GetHashCode();
```

Reply:
??? @lontivero

Reply:
No, this is not okay because two different objects (different internal state) can have the same hashcode and, even when collisions should be minimized,  that's totally okay.